src/components/modal/Modal.svelte

```diff
@@ -0,0 +1,408 @@
+                    {:else}{content}{/if}
+                  </div>
+                </div>
+                <div class="ant-modal-confirm-btns">
+                  {#if modalType === 'confirm'}
+                    <Button
+                      on:click="{onCancelLocal}"
+                      {...cancelButtonProps}
+                      focusOnMount="{autoFocusButton === 'cancel'}">
+                      {cancelText}
+                    </Button>
+                  {/if}
+                  <Button
+                    type="{okType}"
+                    on:click="{onOkLocal}"
+                    {...okButtonProps}
+                    loading="{confirmLoading}"
+                    focusOnMount="{autoFocusButton === 'ok'}">
+                    {okText}
+                  </Button>
+                </div>
+              </div>
+            {/if}
+          </div>
+          {#if footer && modalType === 'standard'}
+            <div class="ant-modal-footer">
+              <slot name="footer">
+                <div>
+                  <Button
+                    on:click="{onCancelLocal}"
+                    {...cancelButtonProps}
+                    focusOnMount="{autoFocusButton === 'cancel'}">
+                    {cancelText}
+                  </Button>
+                  <Button
+                    {...okButtonProps}
+                    type="{okType}"
+                    on:click="{onOkLocal}"
+                    loading="{confirmLoading}"
+                    focusOnMount="{autoFocusButton === 'ok'}">
+                    {okText}
+                  </Button>
+                </div>
+              </slot>
+            </div>
+          {/if}
+        </div>
+      </div>
+    </div>
+  </div>
+{/if}
+
+<script>
+  import { createEventDispatcher, onMount, onDestroy } from "svelte";
+  import { fade, fly } from "svelte/transition";
+  import {
+    CloseOutlined,
+    InfoCircleOutlined,
+    CheckCircleOutlined,
+    CloseCircleOutlined,
+    ExclamationCircleOutlined
+  } from "@/components/icons";
+  import Button from "../button/Button.svelte";
+  import { modalDestroyFunctions } from "./store";
+
+  const dispatch = createEventDispatcher();
+
+  // ********************** Props **********************
+
+  // If the modal is visible
+  export let visible = false;
+  // User has the option to remove the footer
+  export let footer = true;
+  // Modal width
+  export let width;
+  // Title for the header
+  export let title = "";
+  // If the ok button is in loading state
+  export let confirmLoading = false;
+  // Confirm and status modals need a differnet component sructure
+  export let modalType = "standard";
+  // Icon for confirm and status modals
+  export let icon = null;
+  // confirm and status modals can accept content as an option (string or component)
+  export let content = null;
+  // confirm and status modals need to call ok and cancel functions
+  export let onOk = null;
```

Comment:
instead of exporting `onOk` and `onCancel` here I think to make them usable for `Modal.method` we could do something else.

We could get them as options (just like we do now), but instead of passing them to the component we can just do 

```js
modal.$on("ok", () => {
    modal.$set({ visible: false });
    if (options.onOk) {
		options.onOK()
	}
});
modal.$on("cancel", () => {
    modal.$set({ visible: false });
    if (options.onCancel) {
		options.onCancel()
	}
});
```